tests/test_browser.py

```diff
@@ -3347,13 +3347,17 @@ def test_modularize(self):
         self.run_browser('a.html', '...', '/report_result?0')
 
   def test_modularize_network_error(self):
-    src = open(path_from_root('tests', 'browser_test_hello_world.c')).read()
-    create_test_file('test.c', src)
+    test_c_path = path_from_root('tests', 'browser_test_hello_world.c')
     browser_reporting_js_path = path_from_root('tests', 'browser_reporting.js')
-    self.compile_btest(['test.c', '-s', 'MODULARIZE=1', '-s', 'EXPORT_NAME="createModule"', '--extern-pre-js', browser_reporting_js_path])
+    self.compile_btest([test_c_path, '-s', 'MODULARIZE=1', '-s', 'EXPORT_NAME="createModule"', '--extern-pre-js', browser_reporting_js_path])
     create_test_file('a.html', '''
       <script src="a.out.js"></script>
       <script>
+        if (typeof window === 'object') {
+          window.addEventListener('unhandledrejection', function(event) {
+            reportResultToServer("Unhandled promise rejection: " + event.reason.message);
+          });
+        }
```

Comment:
Are the changes to `test_modularize_network_error` just cleanups, and to emit a nicer error, or are they necessary for some reason?

Reply:
I kept the clean up bits and removed the listening to unhandledrejection again.
Ideally, all browser tests should fail on unhandledrejection. But that is better addressed in a separate commit.